### PR TITLE
Port tg github ci config

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
https://github.com/tgstation/tgstation/pull/54668

## Why It's Good For The Game
travis is slow and keeps going down for long periods.

### CODE REVIEW

**Comment by spookydonut** [comment]
i dont know why map configs arent loading or why unit tests arent running

@TiviPlus wanna take a look?

File: tools/deploy.sh [hunk 1]
```diff
@@ -11,27 +11,26 @@ fi
 
 mkdir -p \
     $1/_maps \
-    $1/icons \
-    $1/strings \
-    $1/config
+    $1/icons/runtime \
+    $1/sound/runtime \
+    $1/strings
```
Comment (TiviPlus) [inline tools/deploy.sh:16]: Are you sure about this its causing errors

File: tools/deploy.sh [hunk 2]
```diff
@@ -11,27 +11,26 @@ fi
...
 if [ -d ".git" ]; then
   mkdir -p $1/.git/logs
   cp -r .git/logs/* $1/.git/logs/
 fi
 
 cp tgmc.dmb tgmc.rsc $1/
-cp -r config/* $1/config/
 cp -r _maps/* $1/_maps/
+cp -r icons/runtime/* $1/icons/runtime/
+cp -r sound/runtime/* $1/sound/runtime/
```
Comment (TiviPlus) [inline tools/deploy.sh:26]: ditto

File: tools/deploy.sh [hunk 3]
```diff
@@ -11,27 +11,26 @@ fi
...
 cp -r strings/* $1/strings/
-cp config/maps.txt $1/config/maps.txt
-cp config/shipmaps.txt $1/config/maps.txt
```
Comment (TiviPlus) [inline tools/deploy.sh:28]: Configs?

Comment (spookydonut) [reply tools/deploy.sh:28]: its in run_server.sh

**Pull request merged**